### ISSUE TITLE
Add SOTA classifiers in gpredomicspy for benchmarking

### ISSUE THREAD

**Issue by eprifti** [predomics/gpredomics#53]
## Context

gpredomics focuses on sparse, interpretable models. However, users need to benchmark against state-of-the-art (SOTA) classifiers to demonstrate the value of interpretability vs. raw performance. Implementing these in gpredomicspy provides a unified API for both gpredomics models and SOTA methods, using the same data splits, CV folds, and evaluation metrics.

### Standard ML (scikit-learn)

| Method | Library | Why include |
|--------|---------|-------------|
| **Logistic Regression** (L1/L2/ElasticNet) | sklearn | Interpretable baseline, sparse with L1, closest to gpredomics philosophy |
| **Random Forest** | sklearn | Most common benchmark in metagenomics, handles non-linearities |
| **SVM** (linear + RBF) | sklearn | Strong performer on high-dimensional omics data, linear SVM is interpretable |
| **XGBoost** | xgboost | Gradient boosting SOTA, competitive on tabular data, feature importance |
| **LightGBM** | lightgbm | Faster alternative to XGBoost, handles sparse features natively |

### Regularized / sparse methods

| Method | Library | Why include |
|--------|---------|-------------|
| **Elastic Net** | sklearn | Sparse + grouped selection, good for correlated omics features |
| **Lasso** | sklearn | L1 sparse baseline |
| **SGD Classifier** | sklearn | Scalable linear model, supports multiple loss functions |

### Ensemble / advanced

| Method | Library | Why include |
|--------|---------|-------------|
| **AdaBoost** | sklearn | Classic boosting baseline |
| **Extra Trees** | sklearn | Often outperforms RF on omics data, less variance |
| **Balanced Random Forest** | imbalanced-learn | Handles class imbalance common in clinical cohorts |
| **KNN** | sklearn | Non-parametric baseline |

### Deep learning (optional, lower priority)

| Method | Library | Why include |
|--------|---------|-------------|
| **MLP** | sklearn / PyTorch | Neural net baseline |
| **TabNet** | pytorch-tabnet | Attention-based, interpretable-ish, SOTA on tabular |

## Suggested API

```python
import gpredomicspy as gp

# Run gpredomics
param = gp.Param()
param.load("param.yaml")
exp = gp.fit(param)

# Benchmark against SOTA
benchmark = gp.Benchmark(
    methods=["logistic_l1", "rf", "svm_linear", "xgboost", "lightgbm"],
    cv_folds=5,
    seed=42,
)
results = benchmark.run(X_train, y_train, X_test, y_test)

# Compare with gpredomics
results.add_gpredomics(exp)
results.summary()  # table with AUC, F1, MCC, etc. for all methods
results.plot_roc()  # overlay ROC curves
results.plot_comparison()  # bar chart of metrics
```

## Key design principles

- **Same data splits**: Use identical CV folds for all methods (including gpredomics) for fair comparison
- **Same metrics**: AUC, accuracy, sensitivity, specificity, F1, MCC, PPV, NPV, G-mean — same as gpredomics CSV report
- **Hyperparameter tuning**: Reasonable defaults + optional grid search via sklearn's GridSearchCV
- **Feature importance**: Extract and compare feature importance across methods (permutation importance as common denominator)
- **Export**: Results as pandas DataFrame + CSV, compatible with gpredomics CSV report format

## Implementation notes

- All classifiers are Python-side (scikit-learn / xgboost / lightgbm) — no Rust changes needed
- Data loading can reuse gpredomicspy's existing data loading or accept numpy/pandas directly
- The benchmark should accept the same param.yaml data paths for convenience
- Consider adding prevalence/log transforms as preprocessing options to match gpredomics data types

## Dependencies to add

```
scikit-learn
xgboost
lightgbm
imbalanced-learn (optional)
matplotlib (for plots)
```
